docs/source/python/dataset.rst

```diff
@@ -699,3 +699,46 @@ Parquet files:
 
     # also clean-up custom base directory used in some examples
     shutil.rmtree(str(base), ignore_errors=True)
+
+
+Configuring files open during a write
```

Comment:
I think these sections would actually be best placed next to the "Partitioning performance considerations" section at https://github.com/apache/arrow/blame/master/docs/source/python/dataset.rst#L578

Reply:
I moved the docs to this section.